src/routes/api/product.routes.js

```diff
@@ -28,4 +29,9 @@ productRoutes.post(
 );
 
 productRoutes.get('/', protectRoute, getAllProducts);
+productRoutes.get(
+  '/:productId',
+  protectRoute,
```

Comment:
- add middleware to check if the product exist
- and middleware to check if product is available